### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by jridgewell**
I got carried away with a refactor. Big pictures items are:

- Lots o' tests
- Uses a `Frame` instance for each frame of the stack trace (no more parsing strings, transforming, and joining it back into a string)
- Only CDN js frame sources files are unminified (everything else is left alone, without issuing a request)
- Helpers are extracted out of `routes/error-tracker.js`

### CODE REVIEW

File: routes/error-tracker.js [hunk 1]
```diff
@@ -38,225 +34,145 @@ const SEVERITY = {
...
+  if (version.includes('$internalRuntimeVersion$')) {
+    res.sendStatus(statusCodes.OK);
     return null;
   }
-  if (ignoreMessageOrException(params.m, stack)) {
-    res.set('Content-Type', 'text/plain; charset=utf-8');
-    res.status(statusCodes.BAD_REQUEST);
-    res.send('IGNORE');
+
+  const stack = standardizeStackTrace(params.s || '');
```
Comment (choumx) [inline routes/error-tracker.js]: Sure we want to standardize this stack trace so early? Will probably be throwaway work if we do it before throttling.

File: routes/error-tracker.js [hunk 2]
```diff
@@ -38,225 +34,145 @@ const SEVERITY = {
...
+        referrer: referrer,
       },
     },
   };
   const metaData = {
     resource: {
       type: 'gae_app',
       labels: {
-        version_id: SERVER_START_TIME,
+        version_id: process.env.GAE_VERSION,
```
Comment (choumx) [inline routes/error-tracker.js]: Does this exist during test mode?

File: test/unit/test-debounce.js [hunk 3]
```diff
@@ -14,24 +14,16 @@
...
-const it = mocha.it;
-const describe = mocha.describe;
-const expect = chai.expect;
-const afterEach = mocha.afterEach;
-const beforeEach = mocha.beforeEach;
-const debounce = require('../utils/debounce').debounce;
+const debounce = require('../../utils/debounce');
 
-describe('debounce', function() {
+describe('debounce', () => {
```
Comment (choumx) [inline test/unit/test-debounce.js]: Mocha discourages use of arrow functions to retain access to functions like `this.skip` and `this.timeout`: https://mochajs.org/#arrow-functions

File: utils/cache.js [hunk 4]
```diff
@@ -12,14 +12,16 @@
...
 
-/** Class wraps JS Map object to ensure no entry stays in map more than 2
- *  weeks without retrieval */
+/**
+ * A wrapper around JS Map object to ensure no entry stays in map
+ * more than 2 weeks without retrieval
+ */
 class Cache {
-  /** Create a cache around Map */
+  /** */
```
Comment (choumx) [inline utils/cache.js]: Remove?

File: routes/error-tracker.js (hunk 2 above)
Comment (jridgewell) [inline routes/error-tracker.js]: We can set it, but it's not too important.

File: utils/cache.js (hunk 4 above)
Comment (jridgewell) [inline utils/cache.js]: Needed to pass the linter.

File: test/unit/test-debounce.js (hunk 3 above)
Comment (jridgewell) [inline test/unit/test-debounce.js]: We don't use any of those, yet.

File: routes/error-tracker.js (hunk 1 above)
Comment (jridgewell) [inline routes/error-tracker.js]: Good point, moved throttling higher so we can avoid.

**Pull request merged**